transport_go_test.go

```diff
@@ -97,15 +148,22 @@ func TestTransport_E2E(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	<-clientDone
+	assert.NoError(t, lisClose.Close())
 }
 
-func readLoop(s *BidirectionalStream) {
+func readLoop(s *BidirectionalStream, buf io.Writer, done chan<- struct{}) {
+	var bufSz = 1024
+	buffer := make([]byte, bufSz)
 	for {
-		buffer := make([]byte, 15)
-		_, err := s.ReadInto(buffer)
-		if err != nil {
+		res, err := s.ReadInto(buffer)
+		_, _ = buf.Write(buffer[:res.Amount])
```

Comment:
Maybe just do an `assert.NoError(t, err)`

Even if an error is never returned, just nice to not worry about.